### DISCUSSION TITLE
Users filters not working?

### DISCUSSION THREAD

**Discussion by synapse** [pocketbase/pocketbase#1815] [Q&A] [answered]
Hey @ganigeorgiev , 
I can't seem to make the filters work while requesting the `users` list via API.

While this: 
`http://127.0.0.1:8090/api/collections/users/records?perPage=100` 
seams to be working just fine and retrieves all the users,

this one on the other hand it is not working for me:

`http://127.0.0.1:8090/api/collections/users/records?perPage=1000&filter=(email~"something") `

(although it works perfect from the admin panel).

Am I doing this wrong?

**Comment by ganigeorgiev** [comment, marked as answer]
The `email` field has special behavior and will add an extra condition requiring the auth record email to be visible/public, aka. internally this looks something like:

```
email~"something" && emailVisibility=true
```

(_admins don't have the above special condition; eventually this will be also the case for those auth records with "Manage API" rule access but there were some performance overhead and the "Manage API" rule exception was temporary disabled until a better solution is found_).

**Reply by synapse** [reply]
Ok, thanks @ganigeorgiev, added that small filter option in there and managed to make it work

Thanks again!
Cris

**Reply by ganigeorgiev** [reply]
@synapse I'm not sure that I understand what do you mean. You don't have to manually add the `emailVisibility=true` condition, it is added automatically by the field resolver if an `email` field filter is found. To make the above work, you'll have to update your users `emailVisiblity` field to true (if you want all users to be searchable by email).

**Reply by synapse** [reply]
I haven't had any users with `emailVisibility=true` that's why it would not find them via email, but now it works just fine 👍 .
